mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -5706,3 +5706,95 @@ fn test_electrum_v14_block_hash() {
     // Verify V14 header produces the same hash as our verified BlockHeader implementation
     assert_eq!(hash, headers[0].hash().into());
 }
+
+/// A utility test for debugging block header deserialization issues for any UTXO-based coin.
+/// This test is ignored by default and must be run explicitly.
+///
+/// It scans a range of block heights, fetching headers in chunks. For each chunk, it reads
+/// headers one by one from the data stream.
+///
+/// If it encounters a header that fails to parse, it will panic and print detailed information,
+/// including the exact block height that failed and the raw hex of the entire chunk for context.
+///
+/// # How to Use:
+/// 1.  Modify the constants in the `CONFIGURATION` section below.
+/// 2.  Run the test with the `--ignored` flag: `cargo test -- --ignored test_scan_and_deserialize_block_headers`
+///
+/// # Debugging Note:
+/// If a header at height `N` fails, the error might be caused by the deserializer reading
+/// more data than expected from the header at height `N-1`. The full chunk hex provided
+/// in the panic message is essential for debugging this scenario.
+#[test]
+#[ignore = "This is a utility test for debugging header deserialization and must be run explicitly"]
+fn test_scan_and_deserialize_block_headers() {
+    // ========================== CONFIGURATION ==========================
+    /// The ticker of the coin to test (e.g., "NMC", "CHTA", "RVN").
+    const COIN_TICKER: &str = "NMC";
+    /// A list of active Electrum servers for the specified coin.
+    const ELECTRUM_URLS: &[&str] = &["nmc2.bitcoins.sk:57001", "nmc2.bitcoins.sk:57002"];
+    /// The block height to start scanning from.
+    const START_HEIGHT: u64 = 701614;
+    /// The block height to stop scanning at. Set to `None` to scan to the tip of the chain.
+    const END_HEIGHT: Option<u64> = Some(701616);
+    /// The number of headers to fetch in a single RPC call.
+    const CHUNK_SIZE: u64 = 100;
+    // ===================================================================
+
+    let client = electrum_client_for_test(ELECTRUM_URLS);
+    let mut current_height = START_HEIGHT;
+
+    loop {
+        let mut num_to_fetch = CHUNK_SIZE;
+        if let Some(end_h) = END_HEIGHT {
+            if current_height > end_h {
+                println!("Reached configured end height of {end_h}. Scan complete.");
+                break;
+            }
+            let remaining = end_h.saturating_sub(current_height) + 1;
+            num_to_fetch = num_to_fetch.min(remaining);
+        }
+
+        if num_to_fetch == 0 {
+            break;
+        }
```

Comment:
don't think this is ever gonna trigger.
`remaining` is guaranteed to be greater than `0` and `CHUNK_SIZE` should also not be zero.